### DISCUSSION TITLE
Предложение о добавлении атрибутов

### DISCUSSION THREAD

**Discussion by glebsterx** [IATkachenko/HA-YandexWeather#39] [Ideas]
Предложение:
добавить в атрибуты сущности по-умолчанию `weather.yandex_weather` все, что Яндекс отдает в json, а именно
```
feels_like
temp_water
pressure_mm (для тех кто хочет в mmHg)
wind_gust
...
```
Мотивация: если данные обновляются раз в полчаса -- то отслеживать состояние не имеет большого смысла (имеет конечно, но хз зачем). Держать по сенсору на каждый параметр тоже не всегда удобно, так пусть тогда все данные будут в атрибутах, если это не очень сложно -- их оттуда можно будет вытащить в шаблоны =)

Не знаю сломается ли что-нибудь если вытащить атрибуты еще и прогнозов -- но там тоже было бы можно =)
Спасибо за внимание =)

**Comment by L61senko** [comment]
Присоединяюсь к предложению.
Прошу вынести в атрибуты ключевую часть прогноза погоды (лучше того что через 6 часов)
Конкретно: condition, temperature, templow, возможно precipitation_probability

**Comment by IATkachenko** [comment]
> Присоединяюсь к предложению. Прошу вынести в атрибуты ключевую часть прогноза погоды (лучше того что через 6 часов) Конкретно: condition, temperature, templow, возможно precipitation_probability

Можете мысль развернуть? После #40 основные погодные данные в атрибутах есть. Чего именно не хватает, Прогноза как атрибутов? Сейчас прогноз -- это атрибут `forecast` и все состояния доступны в нем.

**Reply by olesya108** [reply]
В атрибутах прогноза отображается не все, что присылает яндекс:
"part_name": "morning",
"wind_dir": "sw",
"prec_period": 360,

"sunrise": "08:23",
    "sunset": "17:05",
    "moon_code": 3,

- этих, например,  атрибутов нет. Давайте выведем в state все. Спасибо.

**Comment by IATkachenko** [comment]
@olesya108

> В атрибутах прогноза отображается не все, что присылает яндекс: "part_name": "morning", "wind_dir": "sw", "prec_period": 360,
> 
> "sunrise": "08:23", "sunset": "17:05", "moon_code": 3,
> 
>     * этих, например,  атрибутов нет. Давайте выведем в state все. Спасибо.

В том что показывать в прогнозе мы ограничены HA -- в прогнозе можно показать только то что заложено в https://github.com/home-assistant/core/blob/dev/homeassistant/components/weather/__init__.py#L186 

`part_name` -- можно получать из `datetime`
`sunrise`, `sunset` и `moon_code` в прогнозе погоды кажутся лишними и для этого вроде есть более профильные интеграции.
`wind_dir` -- это `wind_bearing` из прогноза (в градусах, со строкой там были какие-то проблемы с отображением)

`prec_period` -- продолжительность осадков единственное что может быть полезно, но в прогноз положить нельзя :(

**Reply by olesya108** [reply]
Хорошо, а что касается part_name - получать ее из datetime не очень удобно. Там по сути timestamp в таймзоне GMT, но в самой строке таймзона не указывается, например - 
2024-02-01T12:40:39.560677

Возможно туда таймзону добавить - 
2024-02-01T06:40:39+00:00
?

**Reply by muaddip01** [reply]
sunrise, sunset и moon_code в прогнозе погоды кажутся лишними и для этого вроде есть более профильные интеграции.

возможно ли добавить эти атрибуты, в связи с закрытием многих сервисов, хотелось бы иметь более стабильный источник данных, сделать как доп. параметр, кому не надо, могут не включать.

**Comment by IATkachenko** [comment]
> Хорошо, а что касается part_name - получать ее из datetime не очень удобно. Там по сути timestamp в таймзоне GMT, но в самой строке таймзона не указывается, например - 2024-02-01T12:40:39.560677
> 
> Возможно туда таймзону добавить - 2024-02-01T06:40:39+00:00 ?

#98, исправлено в [v3.2.1](https://github.com/IATkachenko/HA-YandexWeather/releases/tag/v3.2.1)
